### DISCUSSION TITLE
How deal with response_type: 'token' and Facebook business OAuth

### DISCUSSION THREAD

**Discussion by Untel** [nextauthjs/next-auth#8127] [Help]
### Question 💬

Hi

Following [the facebook login documentation](https://developers.facebook.com/docs/instagram/business-login-for-instagram/), when using `response_type: 'token'` instead of code, the redirection_uri is directly giving the access and long live token.
```
{redirect-url}?
  #access_token={access-token}
  &data_access_expiration_time={data-access-expiration-time}
  &expires_in={expires-in}
  &long_lived_token={long-lived-token}
```
But, i don't know how to access this portion of URL, and how to skip the token request
When i override the token request, params is not giving me the entire redirected URL
```js
token: {
  request: ({ client, params, checks, provider }) {
    // Here we don't need to request the token because the authorization request gave it to us
    console.log({ client, params });
    // {
    //   params: {
    //     client_key: 'xxx',
    //     client_secret: 'xxx',
    //     grant_type: 'authorization_code'
    //     // I should get here the #access_token and long live token
    //   },
    //   client: Client {
    //     authorization_signed_response_alg: 'RS256',
    //     client_id: 'xxx',
    //     client_secret: 'xxx',
    //     grant_types: [
    //       'authorization_code'
    //     ],
    //     id_token_signed_response_alg: 'RS256',
    //     redirect_uris: [
    //       'https://mydomain/api/auth/callback/instagrambusiness'
    //     ],
    //     response_types: [
    //       'code'
    //     ],
    //     token_endpoint_auth_method: 'client_secret_basic'
    //   }
    // }
  }
}
```

Any idea ? 

### How to reproduce ☕️

 [the facebook business login documentation](https://developers.facebook.com/docs/instagram/business-login-for-instagram/)

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by ThangHuuVu** [comment]
hi @Untel this is out of scope for `next-auth` as we only deal with [OAuth Authorization Code Flow](https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow), whereas your link to Business Login for Instagram is a manual flow. You will have to handle this differently.
